core/meterer/meterer.go

```diff
@@ -82,87 +82,66 @@ func (m *Meterer) MeterRequest(ctx context.Context, header core.PaymentMetadata,
-	for quorumID := range reservations {
-		quorumIDs = append(quorumIDs, quorumID)
-		_, protocolConfig, err := params.GetQuorumConfigs(quorumID)
-		if err != nil {
-			return fmt.Errorf("failed to get protocol config for quorum %d: %w", quorumID, err)
-		}
-		reservationWindows[quorumID] = protocolConfig.ReservationRateLimitWindow
-		requestReservationPeriods[quorumID] = GetReservationPeriodByNanosecond(header.Timestamp, protocolConfig.ReservationRateLimitWindow)
-	}
-	// Validate quorumIDs is a subset of the quorumNumbers in the dispersal request; this should be guaranteed by GetReservedPaymentByAccountAndQuorums
-	if err := ValidateQuorum(quorumNumbers, quorumIDs); err != nil {
-		return fmt.Errorf("invalid quorum for reservation: %w", err)
-	}
-
-	// Validate the used reservations are active and is of valid periods
-	for quorumID, reservation := range reservations {
-		if !reservation.IsActiveByNanosecond(header.Timestamp) {
-			return fmt.Errorf("reservation not active")
-		}
-		if !ValidateReservationPeriod(reservation, requestReservationPeriods[quorumID], reservationWindows[quorumID], receivedAt) {
-			return fmt.Errorf("invalid reservation period for reservation on quorum %d", quorumID)
-		}
+// serveReservationRequest handles the rate limiting logic for incoming requests
+func (m *Meterer) serveReservationRequest(ctx context.Context, globalParams *PaymentVaultParams, header core.PaymentMetadata, reservations map[core.QuorumID]*core.ReservedPayment, numSymbols uint64, quorumNumbers []uint8, receivedAt time.Time) error {
+	m.logger.Debug("Recording and validating reservation usage", "header", header, "reservation", reservations)
+	if err := ValidateReservations(reservations, globalParams.QuorumProtocolConfigs, quorumNumbers, header.Timestamp, receivedAt); err != nil {
+		return fmt.Errorf("invalid reservation: %w", err)
 	}
 
 	// Make atomic batched updates over all reservations identified by the same account and quorum
-	if err := m.IncrementBinUsage(ctx, header, reservations, symbolsCharged, reservationWindows, requestReservationPeriods); err != nil {
+	if err := m.incrementBinUsage(ctx, header, reservations, globalParams, numSymbols); err != nil {
 		return fmt.Errorf("failed to increment bin usages: %w", err)
 	}
 	return nil
 }
 
-// IncrementBinUsage increments the bin usage atomically and checks for overflow
-func (m *Meterer) IncrementBinUsage(ctx context.Context, header core.PaymentMetadata, reservations map[core.QuorumID]*core.ReservedPayment, symbolsCharged uint64, reservationWindows map[core.QuorumID]uint64, requestReservationPeriods map[core.QuorumID]uint64) error {
+// incrementBinUsage increments the bin usage atomically and checks for overflow
+func (m *Meterer) incrementBinUsage(ctx context.Context, header core.PaymentMetadata, reservations map[core.QuorumID]*core.ReservedPayment, globalParams *PaymentVaultParams, numSymbols uint64) error {
```

Comment:
same comment, multiple lines for readability
```suggestion
func (m *Meterer) incrementBinUsage(
    ctx context.Context, header core.PaymentMetadata, 
    reservations map[core.QuorumID]*core.ReservedPayment, 
    globalParams *PaymentVaultParams, 
    numSymbols uint64
) error {
```